### PR TITLE
Document how to manipulate attachments via hints

### PR DESCRIPTION
**Pull request by adinauer**
Add some examples on how to add attachments via `Hint`

### CODE REVIEW

File: src/includes/enriching-events/add-attachment/java.mdx [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+import io.sentry.Hint;
+import io.sentry.Sentry;
+import io.sentry.SentryEvent;
+import io.sentry.protocol.Message;
+
+public class MyClass {
+  public void myMethod() {
+    // Add an `EventProcessor` to the current scope. Note that
+    // this helper will process *all* future events in this scope.
+    Sentry.configureScope(scope -> {
```
Comment (marandaneto) [inline src/includes/enriching-events/add-attachment/java.mdx:41]: Maybe don't use `Sentry.configureScope` here but rather focus on the `EventProcessor` class declaration and usage.
You can mention that event processors can be added in the `Scope#addEventProcessor` and `SentryOptions#addEventProcessor`

File: src/includes/enriching-events/add-attachment/java.mdx [hunk 2]
```diff
@@ -0,0 +1,119 @@
+To add an attachment you can either add it to the scope, pass it to any of the `capture` methods, or manipulate the list of attachments in an `EventProcessor` or `beforeSend`.
+
```
Comment (marandaneto) [inline src/includes/enriching-events/add-attachment/java.mdx:2]:
```suggestion
```

One linebreak is enough, fix the others too.

File: src/includes/enriching-events/add-attachment/java.mdx [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+### Adding attachments in beforeSend
+
+Another way of adding attachments is using `beforeSend`:
+
+```java
+import io.sentry.Attachment;
+import io.sentry.Hint;
+import io.sentry.Sentry;
+
+Sentry.init(options -> {
```
Comment (marandaneto) [inline src/includes/enriching-events/add-attachment/java.mdx:96]: That is gonna be visible for Android too but then its `SentryAndroid`.init`, maybe we also add the `android.mdx` or skip the `Sentry.init` block and focus on the `setBeforeSend` callback declaration.

File: src/includes/enriching-events/add-attachment/java.mdx (hunk 1 above)
Comment (marandaneto) [reply src/includes/enriching-events/add-attachment/java.mdx:41]: Also, on Android, there's only one scope anyway.

**Comment by marandaneto** [comment]
I'd also add to the code snippets a way to filter out attachments (removing them)

File: src/includes/enriching-events/add-attachment/java.mdx [hunk 4]
```diff
@@ -0,0 +1,119 @@
+To add an attachment you can either add it to the scope, pass it to any of the `capture` methods, or manipulate the list of attachments in an `EventProcessor` or `beforeSend`.
+
+
+### Passing attachments to capture
```
Comment (imatwawana) [inline src/includes/enriching-events/add-attachment/java.mdx:4]: We use AP style title case for headings in docs

```suggestion
### Passing Attachments to Capture
```

File: src/includes/enriching-events/add-attachment/java.mdx [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+
+    // Send an event to Sentry. During construction of the event
+    // the attachment will be added by the event processor.
+    Sentry.captureMessage("Hello, world!")
+  }
+}
+```
+
+
+### Adding attachments in beforeSend
```
Comment (imatwawana) [inline src/includes/enriching-events/add-attachment/java.mdx:87]:
```suggestion
### Adding Attachments in `beforeSend`
```

File: src/includes/enriching-events/add-attachment/java.mdx [hunk 6]
```diff
@@ -0,0 +1,119 @@
...
+Sentry.init { options ->
+  options.beforeSend = BeforeSendCallback { event, hint ->
+    hint.addAttachment(Attachment("/path/to/file.txt"))
+    event
+  }
+}
+```
+
+
+### Adding attachments to the scope
```
Comment (imatwawana) [inline src/includes/enriching-events/add-attachment/java.mdx:119]:
```suggestion
### Adding Attachments to the Scope
```

File: src/includes/enriching-events/add-attachment/java.mdx [hunk 7]
```diff
@@ -0,0 +1,119 @@
+To add an attachment you can either add it to the scope, pass it to any of the `capture` methods, or manipulate the list of attachments in an `EventProcessor` or `beforeSend`.
+
+
+### Passing attachments to capture
+
+You may pass attachments to any of the `capture` methods, e.g. when capturing an Exception:
```
Comment (imatwawana) [inline src/includes/enriching-events/add-attachment/java.mdx:6]: We avoid latin abbreviations in docs (e.g., i.e., etc, vs)

```suggestion
You may pass attachments to any of the `capture` methods, for example, when capturing an exception:
```

File: src/includes/enriching-events/add-attachment/java.mdx (hunk 1 above)
Comment (adinauer) [reply src/includes/enriching-events/add-attachment/java.mdx]: updated

File: src/includes/enriching-events/add-attachment/java.mdx (hunk 3 above)
Comment (adinauer) [reply src/includes/enriching-events/add-attachment/java.mdx]: updated

**Comment by adinauer** [comment]
> I'd also add to the code snippets a way to filter out attachments (removing them)

Mentioned it as a note now.

Thanks @imatwawana :-)

**Comment by marandaneto** [comment]
@adinauer we can merge this I guess.

**Pull request merged**